numpy/core/tests/test_casting_unittests.py

```diff
@@ -159,6 +183,166 @@ def test_to_void(self):
 
 
 class TestCasting:
+    size = 128
+
+    def get_values(self, dtype, data):
+        length = self.size // dtype.itemsize  # maximum number used
+
+        if dtype.kind == "b":
+            strategy = strategies.booleans()
+        elif dtype.kind in "iu":
+            iinfo = np.iinfo(dtype)
+            strategy = strategies.integers(iinfo.min, iinfo.max)
+        elif dtype.kind == "f":
+            strategy = strategies.floats(width=min(dtype.itemsize * 8, 64))
+        elif dtype.kind == "c":
+            strategy = complex_with_width(width=min(dtype.itemsize * 4, 64))
+        else:
+            raise AssertionError("did not find strategy to fill cast array")
+
+        strategy = strategies.tuples(*([strategy] * length))
+        return data.draw(strategy)
```

Comment:
@Zac-HD just in case you have time to have a look.  These tests work, but are ugly (so I would consider just deleting them if they ever make trouble).

So I am curious if you have an idea for a better solution, I would like to create a single array with "complicated" elements to test against. That could be randomized.  It then ends up casting, which is tricky, because casting is in general tricky... (So hacking the expected casting result is probably worse than the abuse of hypothesis.)

OTOH, it would be cool to improve the tests so to check cast functions for unaligned and strided, etc. inputs conveniently and thoroughly (and maybe also have a blue-print for testing ufuncs better in this regard)